File: client/src/types/file-batch.ts
export interface FileBatch {
  id: string;
  event_id: string;
  label: string;
  file_count: number;
  description: string | null;
  file_types: string | null;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface CreateFileBatchRequest {
  label: string;
  file_count: number;
  description?: string;
  file_types?: string;
  sort_order?: number;
}

export interface UpdateFileBatchRequest {
  label?: string;
  file_count?: number;
  description?: string;
  file_types?: string;
  sort_order?: number;
}

File: server/src/models/file_batch.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.database import Base


class FileBatch(Base):
    __tablename__ = "file_batches"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    event_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("events.id", ondelete="CASCADE"), nullable=False, index=True
    )
    label: Mapped[str] = mapped_column(String(200), nullable=False)
    file_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    file_types: Mapped[str | None] = mapped_column(String(500), nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    # Relationships
    event = relationship("Event", back_populates="file_batches", lazy="selectin")
